src/components/ModelingSidebar/ModelingPanes/FeatureTreePane.tsx

```diff
@@ -0,0 +1,383 @@
+import { Diagnostic } from '@codemirror/lint'
+import { ContextMenu, ContextMenuItem } from 'components/ContextMenu'
+import { CustomIcon, CustomIconName } from 'components/CustomIcon'
+import Loading from 'components/Loading'
+import { useModelingContext } from 'hooks/useModelingContext'
+import { useKclContext } from 'lang/KclProvider'
+import { codeRefFromRange, getArtifactFromRange } from 'lang/std/artifactGraph'
+import { sourceRangeFromRust } from 'lang/wasm'
+import {
+  filterOperations,
+  getOperationIcon,
+  getOperationLabel,
+} from 'lib/operations'
+import { editorManager, engineCommandManager, kclManager } from 'lib/singletons'
+import { reportRejection } from 'lib/trap'
+import { toSync } from 'lib/utils'
+import { ComponentProps, useCallback, useMemo, useRef, useState } from 'react'
+import { Operation } from 'wasm-lib/kcl/bindings/Operation'
+
+export const FeatureTreePane = () => {
+  const { send: modelingSend, state: modelingState } = useModelingContext()
+  // If there are parse errors we show the last successful operations
+  // and overlay a message on top of the pane
+  const parseErrors = kclManager.errors.filter((e) => e.kind !== 'engine')
+
+  // If there are engine errors we show the successful operations
+  // Errors return an operation list, so use the longest one if there are multiple
+  const longestErrorOperationList = kclManager.errors.reduce((acc, error) => {
+    return error.operations && error.operations.length > acc.length
+      ? error.operations
+      : acc
+  }, [] as Operation[])
+
+  const unfilteredOperationList = !parseErrors.length
+    ? !kclManager.errors.length
+      ? kclManager.execState.operations
+      : longestErrorOperationList
+    : kclManager.lastSuccessfulOperations
+
+  // We filter out operations that are not useful to show in the feature tree
+  const operationList = filterOperations(unfilteredOperationList)
+
+  function goToError() {
+    modelingSend({
+      type: 'Set context',
+      data: {
+        openPanes: [...modelingState.context.store.openPanes, 'code'],
+      },
+    })
+    // TODO: this doesn't properly await the set context
+    // so scrolling doesn't work if the code pane isn't open
+    editorManager.scrollToFirstErrorDiagnosticIfExists()
```

Comment:
Maybe make an issue for this?

Reply:
Oh or I could use my `pollUntil` function 😈  or whatever better way there is to do it lol